### ISSUE TITLE
Provide a generic implementation of the RDF test suite 

### ISSUE THREAD

**Issue by MattesWhite** [pchampin/sophia_rs#72]
`sophia` provides traits for parsers and graphs. Accordingly, it should be possible to write a function that automatically runs the [RDF test suite](https://github.com/w3c/rdf-tests/) for the given format.

Following steps are required to do so:

- [x] implement a function to check if two graphs are isomorphic.
   - I think we can borrow the algorithm from [`rio`](https://github.com/Tpt/rio/blob/master/testsuite/src/isomorphism.rs) @Tpt ?
- [ ] write a framework that executes the different test cases.
- [ ] execute tests based on a `manifest`
- [ ] collect test results in a graph.
- [ ] write a function to start evaluation, e.g.
    ```rust
    fn evaluate_parser<P: Parser>(parser_factory: Fn() -> P, directory: Path) -> HashGraph
    ```
    - Of course their should also be a `evaluate_quad_parser()` function.

Except for the isomorphism everything can be feature flagged as it's normally not used.

Having the test suite available would make it easy for implementors to check their crates. In addition, we could analyse crates that have a `sophia`-wrapper and/or implement `sophia::parser::TripleParser`/`QuadParser`, e.g. `rio`.

**Comment by Tpt** [comment]
I have written for Oxigraph [a better version of the isomorphism algorithm than the one provided in Rio](https://github.com/Tpt/oxigraph/blob/2bbe29c31a584ed5c2519a3d28e562b83d8a5882/lib/src/model/isomorphism.rs).

This version does not support named graphs but uses iterative hashing to try to make easier to check when there are a lot of connected blank nodes.

Feel free to reuse it, I spent more than a day on it, it would be very nice to make it useful outside of Oxigraph. Adding back named graphs support should not be hard.

**Comment by MattesWhite** [comment]
I realized that the RDF test suite is rather a use-case of `sophia` than a required part of the library. Therefore, I decided that I'll publish the test suite as an own crate. Accordingly, I suppose to close this issue. 

Once the next version of `sophia` is live I'll publish `rdf-testsuite` (if I'm ready by that time :sweat_smile:). Maybe we could add a _affiliated crates_ section to `sophia`'s `README`?
